### PR TITLE
Add programming interface support

### PR DESCRIPTION
**Pull request by Lukas-Kullmann**
Testcafe's programming is very powerful and lets you handle the tests in a way that is impossible with the CLI tools (e.g. filtering tests with custom functions etc.). So I thought it would be very cool if this was also possible to do with gherkin-testcafe.

### CODE REVIEW

**Comment by Lukas-Kullmann** [comment]
Would also fix #1 since it duplicates #5.

**Comment by Lukas-Kullmann** [comment]
@grebaldi what's the status? Could you take a look?

**Comment by grebaldi** [comment]
Hey @Lukas-Kullmann,

I am so sorry, I didn't take a look at your PR earlier :(

I'm very busy atm and I honestly didn't expect this package to receive that much attention :sweat_smile: 

I took a first look at your work and it really looks great. It's quite an overwhelming contribution, thank you very much!

Unfortunately I won't be able to do a full review just now, but I will get back to your PR today.

Please accept my apologies for the long response time - I know how bad it is having to wait for feedback that long :(

File: package.json [hunk 1]
```diff
@@ -20,11 +20,17 @@
...
+    "endpoint-utils": "^1.0.2",
     "gherkin": "^5.0.0",
     "glob": "^7.1.2",
-    "testcafe": "^0.18.5",
+    "stack-chain": "^2.0.0",
+    "testcafe": "^0.19.2",
     "yargs": "^10.0.3"
+  },
+  "peerDependencies": {
+    "cucumber": "^4.2.1"
```
Comment (grebaldi) [inline package.json]: The least benchmark for this package should be the ability to run 

```
node main.js --specs ./examples/**/*.feature --steps ./examples/**/*.js
```

Just after checking out the source and installing all dependencies.

In order for this to work, `cucumber` needs to be added as a devDependency as well.

File: src/TestcafeGherkinRunner.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const glob = require("glob");
+const path = require("path");
+const TestcafeRunner = require("testcafe/lib/runner/index");
+const TestcafeGherkinBootstrapper = require("./TestcafeGherkinBoostrapper");
```
Comment (grebaldi) [inline src/TestcafeGherkinRunner.js]: There's a typo in the filename for the `TestcafeGherkinBootstrapper` class:

```
+ TestcafeGherkinBootstrapper
- TestcafeGherkinBoostrapper
```

**Review by grebaldi — CHANGES_REQUESTED** [review]
As stated in the code review, I find it crucial that 
```
node main.js --specs ./examples/**/*.feature --steps ./examples/**/*.js
```
works out of the box.

Therefore it would be great, if you could adjust the examples as well, since your approach differs a lot from the status quo there.

Other than these minor issues, your solution looks absolutely great :)

Thanks a lot!

**Comment by Lukas-Kullmann** [comment]
@grebaldi Can you review again, please?

**Comment by grebaldi** [comment]
Hi @Lukas-Kullmann,

sorry for my lack of response. I'm desperately too short on time to maintain this package properly.

Would you be up to take over maintenance for this repo and the associated distributions on docker hub (https://hub.docker.com/r/sitegeist/gherkin-testcafe/) and npm (https://www.npmjs.com/package/gherkin-testcafe)?

**Comment by DennisJaamann** [comment]
Hi @Lukas-Kullmann very interested in using these changes as well, could you open up your fork? I can implement and patch using patch-package from the PR anyway so open sourcing it wouldn't hurt either?

**Comment by Lukas-Kullmann** [comment]
@grebaldi Thanks for the offer. I have to discuss internally, if we are able to maintain this package or how we want to proceed. Can you maybe explain to me, how the docker image is used and why one should use it instead of using local browsers or a provider like browserstack? Internally, we use browserstack and would therefor like to drop the support for the docker image if we would maintain the package.

@DennisJaamann Thanks for showing interest in using our changes. As described above, I have to discuss internally, how we want to proceed with this package in the future. Until then, you can install the package from our fork directly to use our changes: `npm i kiwigrid/gherkin-testcafe` or `yarn add kiwigrid/gherkin-testcafe`.

**Comment by grebaldi** [comment]
@Lukas-Kullmann i think it would be feasible, if you drop support for the docker image.

Since we have a pretty narrow use case for it, I think we can still provide an image ourselves based on your package.

**Comment by Lukas-Kullmann** [comment]
As discussed with @grebaldi, I will take over the maintenance of this package. I just published v1.1.0 from our repo with this PR's changes. Thanks for creating this package @grebaldi!

Since the package will be published from our repo in the future, I close this PR.

@DennisJaamann You can install gherkin-testcafe@1.1 to use our changes. Please report any problems at [our repo](https://github.com/kiwigrid/gherkin-testcafe).

**Comment by DennisJaamann** [comment]
@Lukas-Kullmann great stuff for opening up your fork for the public!

I will launch a PR as soon as I can to support more testcafe CLI options in your fork!

https://github.com/sitegeist/gherkin-testcafe/pull/8

**Pull request closed without merging**